File: src/main/java/org/yearobjectives/infrastructure/database/repository/ObjectiveRepository.java
package org.yearobjectives.infrastructure.database.repository;

import io.quarkus.hibernate.orm.panache.PanacheRepository;
import jakarta.enterprise.context.ApplicationScoped;
import org.yearobjectives.infrastructure.database.entity.ObjectiveEntity;

import java.util.Optional;
import java.util.UUID;

@ApplicationScoped
public class ObjectiveRepository implements PanacheRepository<ObjectiveEntity> {


    public Optional<ObjectiveEntity> findByResourceId(final UUID objectiveId) {
        return find("resourceId", objectiveId).firstResultOptional();
    }
}

File: src/main/kotlin/com/objectives/yearly/infrastructure/database/repository/ObjectiveRepository.kt
package com.objectives.yearly.infrastructure.database.repository

import com.objectives.yearly.infrastructure.database.model.ObjectiveEntity
import com.objectives.yearly.infrastructure.database.model.UserEntity
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import java.util.*

@Repository
interface ObjectiveRepository: JpaRepository<ObjectiveEntity, Long> {

    fun findByUser(currentUser: UserEntity): List<ObjectiveEntity>
    fun findByResourceId(objectiveId: UUID): ObjectiveEntity?

}